server/CGameHandler.cpp

```diff
@@ -1208,25 +1233,12 @@ void CGameHandler::applyBattleEffects(BattleAttack & bat, BattleLogMessage & blm
 		CStack::prepareAttacked(bsa, getRandomGenerator(), bai.defender->acquireState()); //calculate casualties
 	}
 
+	int64_t drainedLife = 0;
 	auto addLifeDrain = [&](int64_t & toHeal, EHealLevel level, EHealPower power)
```

Comment:
Lambda looks a bit overhead since it's too lines (first call already checks toHeal)